docs/components/index.md

```diff
@@ -1,26 +1,156 @@
 <script setup>
-const components = import.meta.glob('./*/data.json', { eager: true })
-const baseUrl = import.meta.env.BASE_URL
+import { ref, computed } from 'vue';
 
-const componentData = Object.keys(components).map(path => {
- return ({
-  ...components[path].default, // component data from JSON
+// Import all JSON data
+const components = import.meta.glob('./*/data.json', { eager: true });
+const baseUrl = import.meta.env.BASE_URL;
+
+// Map JSON data to include href and correct paths
+const componentData = Object.keys(components).map(path => ({
+  ...components[path].default,
   href: `${baseUrl}components/${path.replace('/data.json', '')}`
-})});
+}));
+
+// Reactive variables for query and selected frameworks
+const query = ref('');
+const selectedFrameworks = ref([]);
+
+// Extract unique framework names
+const frameworkNames = computed(() => {
+  const names = new Set();
+  componentData.forEach(component => {
+    component.frameworks.forEach(framework => {
+      names.add(framework.name);
+    });
+  });
+  return Array.from(names);
+});
+
+// Computed property to filter components based on query and selected frameworks
+const filteredComponents = computed(() => {
+  const lowerQuery = query.value.toLowerCase();
+  const frameworks = selectedFrameworks.value;
+
+  return componentData.filter(component => {
+    const matchesQuery = component.title.toLowerCase().includes(lowerQuery);
+    const matchesFramework = frameworks.length === 0 ||
+      component.frameworks.some(framework => frameworks.includes(framework.name));
+    return matchesQuery && matchesFramework;
+  });
+});
 
+// Function to handle button clicks
+const toggleFramework = (framework) => {
+  if (selectedFrameworks.value.includes(framework)) {
+    selectedFrameworks.value = selectedFrameworks.value.filter(f => f !== framework);
+  } else {
+    selectedFrameworks.value.push(framework);
+  }
+};
+
+// Function to handle "All" button click
+const showAll = () => {
+  selectedFrameworks.value = [];
+};
 </script>
 
+<style>
+  /* remove these to custom.css and replace with warp-doc color tokens */
```

Comment:
Maybe we could add `TODO` so it's easier to check what stuff needs to be fixed when the time comes? Also, what about adding a reference to who added it wherever stuff needs to be moved, removed etc.? Something like `TODO(@Skadefryd): remove these to custom.css and replace with warp-doc color tokens`. I'm mostly thinking about a scenario where we have to move the whole repo contents to another repo and it's hard to know who added the todos in case we need to ask for more info.